.github/workflows/build.yaml

```
@@ -1,5 +1,8 @@
 name: Build Notebook Container
-on: push
+on:
+  - push
+  - pull_request
```

There are plenty of ways to declare the same thing. By declaring it like I suggest below it is easier to understand how to add details to the configuration going onwards.

```suggestion
on:
  push:
  pull_request:
  workflow_dispatch:
```

It is reasonable for example to include an `ignore-paths` check here to ignore `.circleci` folder or similar. I also like including `workflow_dispatch` allowing you to trigger tests manually with a "Run workflow" button.

---

I'm a bit confused about the image pushing that is being done in the `build` job.

The "Setup repo2docker" step part of the `build` job is doing a lot of things, including pushing to a docker registry it seems.

I see we use a dummy image name etc so I assume no credentials are setup for this, but if there were credentials setup it would be trouble to use a pull_request trigger as those wouldn't be accessible then.

Is the `test` image special for dockerhub?

---

nah, this is a total prototype as i feel around to see how things are :)

I'll just move this repo to my personal account to confuse people less.

+1 on using dict forms than list forms for `on:` tho